ui/components/app/signature-request-original/signature-request-original.component.js

```diff
@@ -400,10 +409,40 @@ export default class SignatureRequestOriginal extends Component {
           <SignatureRequestOriginalWarning
             senderAddress={address}
             name={name}
-            onSubmit={async (event) => await this.onSubmit(event)}
+            onSubmit={async () => {
+              ///: BEGIN:ONLY_INCLUDE_IF(build-flask)
+              if (warnings?.length >= 1) {
+                return this.setState({
+                  showSignatureInsights: true,
+                  showSignatureRequestWarning: false,
+                });
+              }
+              ///: END:ONLY_INCLUDE_IF
+              return await this.onSubmit();
+            }}
             onCancel={async (event) => await this.onCancel(event)}
           />
         )}
+        {
+          ///: BEGIN:ONLY_INCLUDE_IF(build-flask)
+        }
+        {this.state.showSignatureInsights && (
+          <InsightWarnings
+            warnings={warnings}
+            action="signing"
+            origin={txData.msgParams.origin}
+            onCancel={() => {
+              this.setState({ showSignatureInsights: false });
+            }}
+            onSubmit={async () => {
+              await this.onSubmit();
+              this.setState({ showSignatureInsights: false });
+            }}
+          />
+        )}
```

Comment:
Logic around setting `showSignatureInsights` can be moved into `InsightWarnings` or another component wrapping `InsightWarnings`.

This will avoid adding this state handling to all signature request pages individually.

Reply:
I had decided to keep logic at the signature request page levels because some of the logic is dependent on other state such as `showSignatureRequestWarning` and didn't want to introduce more complexity.